tool/tsh/access_request_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+func newKeygen(t *testing.T, precompute int) *native.Keygen {
+	fakeClock := clockwork.NewFakeClockAt(time.Date(2016, 9, 8, 7, 6, 5, 0, time.UTC))
+
+	keygen := native.New(
+		context.TODO(),
+		native.PrecomputeKeys(precompute),
+		native.SetClock(fakeClock))
+
+	t.Cleanup(func() { keygen.Close() })
+
+	return keygen
+}
+
+func newKey(keygen *native.Keygen, roles []string) (*client.Key, error) {
+	priv, pub, err := keygen.GenerateKeyPair("")
+	if err != nil {
+		return nil, err
+	}
+	cert, err := keygen.GenerateUserCertWithoutValidation(
+		services.UserCertParams{
+			PrivateCASigningKey:   priv,
+			CASigningAlg:          defaults.CASignatureAlgorithm,
+			PublicUserKey:         pub,
+			Username:              "some-user",
+			AllowedLogins:         []string{},
+			TTL:                   time.Hour,
+			Roles:                 roles,
+			CertificateFormat:     teleport.CertificateFormatStandard,
+			PermitAgentForwarding: true,
+			PermitPortForwarding:  true,
+		})
+	if err != nil {
+		return nil, err
+	}
+
+	key := &client.Key{
+		Priv: priv,
+		Pub:  pub,
+		Cert: cert,
+	}
+
+	return key, nil
+}
+
+func TestPassingNilToIsNotFoundReturnsFalse(t *testing.T) {
+	require.False(t, trace.IsNotFound(nil))
+}
```

Comment:
was this for debugging?

Reply:
Just asserting something I wasn't sure of.

Reply:
Deleted